rest_framework/relations.py

```diff
@@ -58,6 +60,16 @@ def __init__(self, *args, **kwargs):
 
     def initialize(self, parent, field_name):
         super(RelatedField, self).initialize(parent, field_name)
+
+        if self.queryset is None and not self.limit_choices_to:
+            manager = getattr(self.parent.opts.model, self.source or field_name)
+            try:
+                self.limit_choices_to = manager.field.rel.limit_choices_to
+            except AttributeError:
+                # GenericForeignKey and their reverse relationships don't have
+                # a `field` property.
+                pass
+
```

Comment:
I'm not sure what this code block is for. `self.limit_choices_to` is only relevant when `queryset is not None` so why do we need to set it to anything here?